mqtt-service/model/src/main/java/com/cumulocity/mqtt/service/client/model/MqttMetadata.java

```diff
@@ -103,6 +103,8 @@ public String getResponseTopic() {
     }
 
     /**
+     * Retrieves the MQTT topic. If necessary, this property must be explicitly configured when employing an MQTT WebSocket Producer.
```

Comment:
What do you think about rephrasing it to something like:
```
Represents MQTT topic name when receiving messages via {@link MqttMessageListener}. When publishing the message via {@link MqttPublisher.publish} this field is currently ignored.
```

Reply:
I don't know if that makes it clear. 
The concern was that it does not say this field is not automatically set in case of using WS producer and consuming messages from WS consumer.

Reply:
You're kind of both right :-)  My concern was that you can only rely on this field when it was set by the MQTT engine, on a message published by an MQTT device.  In that case it does reflect the topic that the message was published to.  However, it is essentially meaningless for messages published by a microservice, through the SDK.

Can we put Javadoc on the generated constructor?  As a user of the SDK I would want to know that setting this field does not affect the MQTT topic that a message will actually be published on (it would be reasonable for a user to assume that it does control the topic).  I would also want to know that the value will be passed through unchanged to any SDK consumers that receive the message (I think that is true?) which could be confusing if it's set to a different value than the actual topic.

On the getter side, shouldn't we just say:
1. For messages published by an MQTT client, this field contains the name of the topic that the message was published to.
2. For messages published by an SDK client, this field contains the value passed to the MqttMetadata constructor, which is not necessarily the same as the MQTT topic the message was published to.

Sorry for the long comment!